lib/elixir/lib/option_parser.ex

```diff
@@ -394,16 +394,31 @@ defmodule OptionParser do
       ["--bool", "--no-bool"]
 
   """
-  @spec to_argv(Enumerable.t) :: argv
-  def to_argv(enum) do
+  @spec to_argv(Enumerable.t, options) :: argv
+  def to_argv(enum, opts \\ []) do
     Enum.flat_map(enum, fn
       {_key, nil}  -> []
       {key, true}  -> [to_switch(key)]
       {key, false} -> [to_switch(key, "--no-")]
-      {key, value} -> [to_switch(key), to_string(value)]
+      {key, value} -> to_argv(key, value, opts)
     end)
   end
 
+  # If no options were provided consider the argv a simple pair
+  defp to_argv(key, value, []) do
```

Comment:
We can remove this clause and add it directly to the `else` branch of the clause below.